src/main/java/com/commercetools/sync/products/utils/ProductReferenceResolutionUtils.java

```diff
@@ -221,56 +225,13 @@ static CategoryReferencePair mapToCategoryReferencePair(@Nonnull final Product p
   }
 
   /**
-   * Builds a {@link ProductQuery} for fetching products from a source CTP project with all the
-   * needed references expanded for the sync:
+   * Builds a {@link ProductQuery} for fetching products from a source CTP project.
    *
-   * <ul>
-   *   <li>Product Type
-   *   <li>Tax Category
-   *   <li>Product State
-   *   <li>Staged Assets' Custom Types
-   *   <li>Staged Product Categories
-   *   <li>Staged Prices' Channels
-   *   <li>Staged Prices' Customer Groups
-   *   <li>Staged Prices' Custom Types
-   *   <li>Reference Attributes
-   *   <li>Reference Set Attributes
-   * </ul>
-   *
-   * <p>Note: Please only use this util if you desire to sync all the aforementioned references from
-   * a source commercetools project. Otherwise, it is more efficient to build the query without
-   * expansions, if they are not needed, to avoid unnecessarily bigger payloads fetched from the
-   * source project.
-   *
-   * @return the query for fetching products from the source CTP project with all the aforementioned
-   *     references expanded.
+   * @return the query for fetching products from the source CTP project.
    */
   @Nonnull
   public static ProductQuery buildProductQuery() {
-    return ProductQuery.of()
-        .withLimit(QueryExecutionUtils.DEFAULT_PAGE_SIZE)
-        .withExpansionPaths(ProductExpansionModel::productType)
-        .plusExpansionPaths(ProductExpansionModel::taxCategory)
-        .plusExpansionPaths(ExpansionPath.of("state"))
-        .plusExpansionPaths(expansionModel -> expansionModel.masterData().staged().categories())
-        .plusExpansionPaths(
-            expansionModel -> expansionModel.masterData().staged().allVariants().prices().channel())
-        .plusExpansionPaths(
-            expansionModel ->
-                expansionModel.masterData().staged().allVariants().prices().customerGroup())
-        .plusExpansionPaths(
-            ExpansionPath.of("masterData.staged.masterVariant.prices[*].custom.type"))
-        .plusExpansionPaths(ExpansionPath.of("masterData.staged.variants[*].prices[*].custom.type"))
-        .plusExpansionPaths(
-            expansionModel ->
-                expansionModel.masterData().staged().allVariants().attributes().value())
-        .plusExpansionPaths(
-            expansionModel ->
-                expansionModel.masterData().staged().allVariants().attributes().valueSet())
-        .plusExpansionPaths(
-            ExpansionPath.of("masterData.staged.masterVariant.assets[*].custom.type"))
-        .plusExpansionPaths(
-            ExpansionPath.of("masterData.staged.variants[*].assets[*].custom.type"));
+    return ProductQuery.of();
```

Comment:
That would conflict with Andi's work, please consider not merging those changes into master until the whole development is done.

Reply:
Sure, I will wait for Andi to merge his changes into master and then i will fix the conflicts in my branch.

Reply:
you might base to `release_5` branch, that Andi created already, so we could review this again, as there would be lots of conflict changes

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/697/commits/4890668f7c7cf19bfab7e7ed41860fdbc8a67b3f